docs/src/api/class-route.md

```diff
@@ -386,28 +386,115 @@ affect the route matching, all the routes are matched using the original request
 * since: v1.23
 - `method` <[string]>
 
-If set changes the request method (e.g. GET or POST)
+If set changes the request method (e.g. GET or POST).
 
 ### option: Route.fallback.postData
 * since: v1.23
 * langs: js, python, java
 - `postData` <[string]|[Buffer]>
 
-If set changes the post data of request
+If set changes the post data of request.
 
 ### option: Route.fallback.postData
 * since: v1.23
 * langs: csharp
 - `postData` <[Buffer]>
 
-If set changes the post data of request
+If set changes the post data of request.
 
 ### option: Route.fallback.headers
 * since: v1.23
 - `headers` <[Object]<[string], [string]>>
 
 If set changes the request HTTP headers. Header values will be converted to a string.
 
+## async method: Route.fetch
+* since: v1.29
+- returns: <[APIResponse]>
+
+Performs the request and fetches result without fulfilling it, so that the response
+could be modified and then fulfilled.
+
+**Usage**
+
+```js
+await page.route('https://dog.ceo/api/breeds/list/all', async route => {
+  const response = await route.fetch();
+  const json = await response.json();
+  json.message['big_red_dog'] = [];
+  await route.fulfill({ response, json });
+});
+```
+
+```java
+page.route("https://dog.ceo/api/breeds/list/all", route -> {
+  APIResponse response = route.fetch();
+  JsonObject json = new Gson().fromJson(response.text(), JsonObject.class);
+  json.set("big_red_dog", new JsonArray());
+  route.fulfill(new Route.FulfillOptions()
+    .setResponse(response)
+    .setJson(json.toString()));
```

Comment:
`setBody` ? there is no point in setJson in java since we don't support any particular json lib.